spring-xd-dirt/src/main/java/org/springframework/xd/dirt/rest/metrics/AbstractMetricsController.java

```diff
@@ -66,8 +67,28 @@ public AbstractMetricsController(R repository) {
 	}
 
 	/**
+	 * Lists metric resources.
+	 *
+	 * @param pageable
+	 * @param pagedAssembler
+	 * @param resourceAssembler
+	 * @return
+	 */
+	protected PagedResources<? extends MetricResource> list(Pageable pageable,
+			PagedResourcesAssembler<M> pagedAssembler,
+			ResourceAssemblerSupport<M, ? extends MetricResource> resourceAssembler) {
+		/* Page */Iterable<M> metrics = repository.findAll(/* pageable */);
+
+		// Ok for now until we use PagingAndSortingRepo as we know we have lists
+		Page<M> page = new PageImpl<M>((List<M>) metrics);
+		return pagedAssembler.toResource(page,
```

Comment:
I'd change this to always require a third argument and have the subclass dynamically pass the correct assembler (`shallowResourceAssembler` will need to become protected then)